Source/Schema.NET/Thing.Partial.cs

```diff
@@ -34,8 +35,35 @@ public partial class Thing : JsonLdObject
             },
             DefaultValueHandling = DefaultValueHandling.Ignore,
             NullValueHandling = NullValueHandling.Ignore,
+            DateParseHandling = DateParseHandling.DateTimeOffset,
             StringEscapeHandling = StringEscapeHandling.EscapeHtml
         };
+        
+        
+        /// <summary>
+        /// Serializer settings used when deserializing.
+        /// </summary>
+        private static readonly JsonSerializerSettings DeserializerSettings = new JsonSerializerSettings()
+        {
+            Converters = new List<JsonConverter>()
+            {
+                new StringEnumConverter()
+            },
+            NullValueHandling = NullValueHandling.Ignore,
+            DefaultValueHandling = DefaultValueHandling.Ignore,
+            DateParseHandling = DateParseHandling.DateTimeOffset,
+            // The ASP.NET MVC framework defaults to 32 (so 32 levels deep in the JSON structure)
+            // to prevent stack overflow caused by malicious complex JSON requests.
+            MaxDepth = 32
+        };
+
+        /// <summary>
+        /// Returns a strongly typed model of the JSON-LD representation provided.
+        /// </summary>
+        /// <typeparam name="T">Type of schema.org object to deserialize (can use Thing for any)</typeparam>
+        /// <param name="str">JSON string</param>
+        /// <returns>Strongly typed schema.org model</returns>
+        public static T Deserialize<T>(string str) => JsonConvert.DeserializeObject<T>(str, DeserializerSettings);

```

Comment:
This is a reasonable addition. If we do this, then we should add an overload that takes a stream. Also, we should document this addition in the main README.md file.

Reply:
Great idea!

Reply:
Also on this line - we should also add a constraint of `where T : Schema.NET.JsonLdObject` (see https://github.com/openactive/OpenActive.NET/blob/master/OpenActive.NET/OpenActiveSerializer.cs#L125 for example)